### PR TITLE
Release v0.7.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v0.7.0

Autonomy:
- Adds support for updating the participants list at the runtime using the on chain metadata of a service
- Enables the usage of `gRPC` channel when communicating with the `abci` connection
- Updates the tendermint image name constant to make sure it's up to date with the latest framework version

Packages:
- Adds validation for the setup data and raises early if necessary data are not provided instead of waiting for the first round to happen
- Removes the fast forward round because it is not meaningful since the setup data cannot be empty
- Moves `all_participants` to the setup data
- Enables the usage of `gRPC` channel when communicating with the tendermint node
- Adds meta classes for the `AbstractRound` and the `BaseBehaviour` classes to enforce additional checks
- Fixes typing issues on the synchronized database class
- Removes the inappropriate usage of the `# type: ignore` marker and addresses the typing issues properly

Tests:
- Test coverage for `BaseTestEnd2End` in the test plugin
- Test coverage for the `abci` connection
- Test coverage for tendermint protocol dialogues
- Updates hello world `e2e` test to test the usage of `gRPC` channel on tendermint node
- Fixes the inconsistencies regarding the usage of `setup` and `setup_class` methods in the test classes
- Introduces base class for testing test tools
- Adds tests for transaction settlement integration test tools

Docs:
- Updates the guide for running a service on different networks
- Reorganizes the developer tooling section

Chores:
- Updates the coverage collection in the CI to aggregate the coverage for both framework and the packages

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] I've updated the dependencies versions to the latest, wherever is possible.
- [ ] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ Below we describe the additional manual steps required to upgrade between differ
 
 # Open Autonomy
 
+## `v0.6.0` to `v0.7.0`
+
+No backwards incompatible changes
```
Comment (0xArdi) [inline docs/upgrading.md:10]: Why a major release then?

Comment (angrybayblade) [reply docs/upgrading.md:10]: Sorry, I forgot about the comment. I think there were some breaking changes, I think either @DavidMinarsch or @Adamantios can help here

File: docs/upgrading.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ Below we describe the additional manual steps required to upgrade between differ
...
+
```
Comment (DavidMinarsch) [inline docs/upgrading.md:11]:
```suggestion

The new `AbstractRound` and `BaseBehaviour` metaclasses enforce checks on the class attributes. For concrete classes inheriting from `AbstractRound` the developer must set `synchronized_data_class`, `allowed_tx_type` and `payload_attribute`. For concrete classes inheriting from `BaseBehaviour` the developer must set `matching_round`. These assumptions were present before but not enforced at the class-definition level by the framework.

The set of `all_participants` is now retrieved from the safe instance referenced by `safe_contract_address` and assumed to be present on the target chain. Use the dev tools for running a local development image with pre-configured safe.
```

File: docs/upgrading.md (hunk 1 above)
Comment (DavidMinarsch) [reply docs/upgrading.md:10]: See below. @Adamantios might be able to improve it further.

**Review by DavidMinarsch — APPROVED** [review]

**Review by Adamantios — APPROVED** [review]

**Pull request merged**